zkevm-circuits/src/evm_circuit/util/instrumentation.rs

```diff
@@ -0,0 +1,170 @@
+                        report.storage_2 = data_entry;
+                    }
+                    CellType::StoragePermutation => {
+                        report.storage_perm = data_entry;
+                    }
+                    CellType::LookupByte => {
+                        report.byte_lookup = data_entry;
+                    }
+                    CellType::Lookup(Table::Fixed) => {
+                        report.fixed_table = data_entry;
+                    }
+                    CellType::Lookup(Table::Tx) => {
+                        report.tx_table = data_entry;
+                    }
+                    CellType::Lookup(Table::Rw) => {
+                        report.rw_table = data_entry;
+                    }
+                    CellType::Lookup(Table::Bytecode) => {
+                        report.bytecode_table = data_entry;
+                    }
+                    CellType::Lookup(Table::Block) => {
+                        report.block_table = data_entry;
+                    }
+                    CellType::Lookup(Table::Copy) => {
+                        report.copy_table = data_entry;
+                    }
+                    CellType::Lookup(Table::Keccak) => {
+                        report.keccak_table = data_entry;
+                    }
+                    CellType::Lookup(Table::Exp) => {
+                        report.exp_table = data_entry;
+                    }
+                }
+            }
+            report_collection.push(report);
+        }
+        report_collection
+    }
+}
+
+/// Struct which contains a Cost/ColumnType report for a particular EVM
+/// `ExecutionStep`.
+#[derive(Clone, Debug, Default)]
+pub(crate) struct ExecStateReport {
+    pub(crate) state: ExecutionState,
+    pub(crate) storage_1: StateReportRow,
+    pub(crate) storage_2: StateReportRow,
+    pub(crate) storage_perm: StateReportRow,
+    pub(crate) byte_lookup: StateReportRow,
+    pub(crate) fixed_table: StateReportRow,
+    pub(crate) tx_table: StateReportRow,
+    pub(crate) rw_table: StateReportRow,
+    pub(crate) bytecode_table: StateReportRow,
+    pub(crate) block_table: StateReportRow,
+    pub(crate) copy_table: StateReportRow,
+    pub(crate) keccak_table: StateReportRow,
+    pub(crate) exp_table: StateReportRow,
+}
+
+impl From<ExecutionState> for ExecStateReport {
+    fn from(state: ExecutionState) -> Self {
+        ExecStateReport {
+            state,
+            ..Default::default()
+        }
+    }
+}
+
+impl From<&ExecutionState> for ExecStateReport {
+    fn from(state: &ExecutionState) -> Self {
+        ExecStateReport {
+            state: *state,
+            ..Default::default()
+        }
+    }
+}
+
+/// Struct that contains all of the measurament values required to evaluate the
+/// costs of a particular `ColumnType` of an `ExecStateReport`
+#[derive(Debug, Clone, Default)]
+pub(crate) struct StateReportRow {
+    pub(crate) avaliable_cells: usize,
```

Comment:
1. Found a typo
2. Can we doc these fields?

```suggestion
    pub(crate) available_cells: usize,
```